src/main/java/io/mosip/mimoto/controller/CredentialsController.java

```diff
@@ -79,4 +82,33 @@ public ResponseEntity<?> downloadCredentialAsPDF(
         }
     }
 
+    @PostMapping("/verify")
+    public ResponseEntity<?>verifyCredential(@RequestBody String credential){
+
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v1");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+
+        try{
+            Boolean verificationResult = credentialService.verifyCredential(credential);
+            responseWrapper.setResponse(verificationResult);
+            return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+
+        }catch(ProofDocumentNotFoundException exception){
+            responseWrapper.setErrors(List.of(new ErrorDTO(PROOF_DOCUMENT_NOT_FOUND_EXCEPTION.getCode(), exception.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        } catch (ProofTypeNotFoundException exception) {
+            responseWrapper.setErrors(List.of(new ErrorDTO(PROOF_TYPE_NOT_FOUND_EXCEPTION.getCode(), exception.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        } catch (SignatureVerificationException exception) {
+            responseWrapper.setErrors(List.of(new ErrorDTO(SIGNATURE_VERIFICATION_EXCEPTION.getCode(), exception.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        }catch (UnknownException exception) {
+            responseWrapper.setErrors(List.of(new ErrorDTO(UNKNOWN_EXCEPTION.getCode(), exception.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
```

Comment:
return statement is common in all catch exceptions.
It can be moved to finally statement

Reply:
Fix spacing and indentation